```{r setup, cache=FALSE, echo=FALSE, results='asis'}
dumpcssfile <- function(fname) {
  paste(c('<style type="text/css">', readLines(fname), '</style>\n'),
        collapse="\n")
}

opts_chunk$set(cache=TRUE,
               autodep=TRUE,
               fig.align="center",
               comment="")

knit_hooks$set(error=function(x, options) stop(x),
               fig.cap=function(before, options, envir) {
                 if (!before) {
                   paste0('<p class="caption">', options$fig.cap, "</p>")
                 }
               })

cat(dumpcssfile(file.path("css", "ieo.css")))
```

# Analysis of a TCGA RNA-seq data set on Breast invasive carcinoma

### Ainoha Garcia (@upf.edu), Joan F. Gilabert (joanfrancesc.gilabert01@estudiant.upf.edu) and Leo (@upf.edu)

## Introduction
Breast invasive carcinoma (BRCA), is the most common malignant cancer affecting women and is the second leading cause of cancer death worldwide. This disease has more than 1,300,000 cases and 450,000 death each year around the world[3]. This disease is widely heterogeneous, having a large and diverse set of molecular, histological and clinical behaviors depending of the tumour. The Cancer Genome Atlas (TCGA) has comprehensively profiled this type of cancer in a patient cohort. Here we analyze the expression profiles of those patients, accessible in the form of a raw RNA-seq counts produced by
[Rahman et al. (2015)](http://bioinformatics.oxfordjournals.org/content/31/22/3666.short) using a pipeline based on the R/Bioconductor software package [Rsubread](http://bioconductor.org/packages/Rsubread).

## Data import

We start importing the raw table of counts.

```{r}
library(SummarizedExperiment)

se <- readRDS(file.path("data/seBRCA.rds"))
se
```

Explore the column (phenotypic) data, which in this case corresponds to clinical
variables, and their corresponding metadata.

```{r}
dim(colData(se))
colData(se)[1:5, 1:5]
mcols(colData(se), use.names=TRUE)
```
These metadata consists of two columns of information about the clinical variables.
One called `labelDescription` contains a succinct description of the variable, often
not more self-explanatory than the variable name itself, and the other called
'CDEID' corresponds to the so-called `Common Data Element (CDE)` identifier. This
identifier can be use in https://cdebrowser.nci.nih.gov to search for further
information about the associated clinical variable using the `Advanced search`
form and the `Public ID` attribute search.

Now, explore the row (feature) data.

```{r}
rowRanges(se)
```

This feature data provides information about the genes, their symbols, location in the genome, length and gc content.

## Creation of subsets

We focus on a subset of our data to perform the analysis. To create this subset we use a paired data criterion. That is, we keep only the data corresponding to patient that have samples of tumor and normal type. We use this criterion in order the minimize the genomic variability and supposedly reduce the probability of having confounding factors in our data. The following code performs this filtering:

```{r}
codesnorm <- colData(se)[colData(se)$type == "normal",]$bcr_patient_barcode
codestum <- colData(se)[colData(se)$type == "tumor",]$bcr_patient_barcode
commoncodes <- codesnorm[codesnorm %in% codestum]
se <- se[,colData(se)$bcr_patient_barcode %in% commoncodes]
dim(se)
table(se$type)
```

As we can see our filtered data-set consists now of 20115 genes for 212 samples, half of which are tumor type and the other half are normal samples.

## Quality assessment and normalization

To perform quality assessment and normalization we need first to load the
[edgeR](http://bioconductor.org/packages/edgeR) R/Bioconductor package and
create a `DGEList' object.

```{r}
library(edgeR)
names(se) <- rowRanges(se)$symbol
dge <- DGEList(counts=assays(se)$counts, genes=mcols(se))
```
Now calculate $\log_2$ CPM values of expression and put them as an additional
assay element to ease their manipulation.

```{r}
assays(se)$logCPM <- cpm(dge, log=TRUE, prior.count=0.5)
assays(se)$logCPM[1:5, 1:5]
```
### Library sizes

Let's examine the library sizes in terms of total number of sequence read counts
per sample. Figure S1 below shows library sizes per sample in increasing order.

```{r libsizes, echo=FALSE, out.width="600px", fig.cap="Figure S1: Library sizes in increasing order."}
ord <- order(dge$sample$lib.size/1e6)
barplot(dge$sample$lib.size[ord]/1e6, las=1, ylab="Millions of reads",
        xlab="Samples", col=c("blue", "red")[(se$type[ord] == "tumor") + 1], cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5)
legend("topleft", c("tumor", "normal"), fill=c("red", "blue"), inset=0.01, cex=1.5)
```

This figure reveals substantial differences in sequencing depth between samples, although this differences are quite homogeneously distributed between normal and tumor samples. We might filter the samples that have a library size with an extreme value, although that would disrupt our paired subset, so we decided to leave all samples. 

### Distribution of expression levels among samples

Let's look at the distribution of expression values per sample in terms of
logarithmic CPM units. Due to the large number of samples, we display tumor
and normal samples separately, and are shown in Figure S2.

```{r distRawExp, echo=FALSE, fig.height=4, fig.width=10, out.width="800px", fig.cap="Figure S2: Non-parametric density distribution of expression profiles per sample."}
library(geneplotter)
par(mfrow=c(1, 2))
multidensity(as.list(as.data.frame(assays(se[, se$type == "tumor"])$logCPM)),
             xlab="log 2 CPM", legend=NULL, main="Tumor samples", las=1)
multidensity(as.list(as.data.frame(assays(se[, se$type == "normal"])$logCPM)),
             xlab="log 2 CPM", ylim=c(0.00,0.15), legend=NULL, main="Normal samples", las=1)
```
In the figure we do not appreciate remarkable differences between the distribution of tumor samples and normal samples.

### Distribution of expression levels among genes

Let's calculate now the average expression per gene through all the samples. Figure S3
shows the distribution of those values across genes.

```{r exprdist, echo=FALSE, out.width="400px", fig.cap="Figure S3: Distribution of average expression level per gene."}
par(mar = c(6.5, 6.5, 0.5, 0.5), mgp = c(5, 1, 0))
avgexp <- rowMeans(assays(se)$logCPM)
hist(avgexp, xlab="log2 CPM", main="", las=1, cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5)
abline(v=1, col="red", lwd=2)
```

### Filtering of lowly-expressed genes

In the light of this plot, we may consider a cutoff of 1 log CPM unit as minimum value
of expression to select genes being expressed across samples. Using this cutoff we proceed
to filter out lowly-expressed genes.

```{r}
mask <- avgexp > 1
dim(se)
dim(dge)
se <- se[mask, ]
dge <- dge[mask, ]
dim(se)
dim(dge)
```

### Normalization

We calculate now the normalization factors on the filtered expression data set.

```{r}
dge <- calcNormFactors(dge)
```

Replace the raw log2 CPM units in the corresponding assay element of the `SummarizedExperiment`
object, by the normalized ones.

```{r}
assays(se)$logCPM <- cpm(dge, log=TRUE, prior.count=0.5)
```

### MA-plots

We examine now the MA-plots of the normalized expression profiles. We look first to
the tumor samples.
```{r maPlotsTumor, fig.height=100, fig.width=16, dpi=100, echo=FALSE, fig.cap="Figure S4: MA-plots of the tumor samples."}
par(mfrow=c(36, 3), mar=c(4, 5, 3, 1))
setmp <- se[, se$type == "tumor"]
dgetmp <- dge[, se$type == "tumor"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1, cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```
We do not observe samples with major expression-level dependent biases. Let's look now to the normal samples.

```{r maPlotsNormal, fig.height=100, fig.width=16, dpi=100, echo=FALSE, fig.cap="Figure S5: MA-plots of the normal samples."}
par(mfrow=c(36, 3), mar=c(4, 5, 3, 1))
setmp <- se[, se$type == "normal"]
dgetmp <- dge[, se$type == "normal"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1, cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

Again, we do not observe either important expression-level dependent biases among the normal samples.

### Batch identification

We will search now for potential surrogate of batch effect indicators. Given that each sample
names corresponds to a TCGA barcode (see https://wiki.nci.nih.gov/display/TCGA/TCGA+barcode),
following the strategy described in http://bioinformatics.mdanderson.org/main/TCGABatchEffects:Overview
we are going to derive different elements of the TCGA barcode and examine their distribution
across samples.

```{r}
tss <- substr(colnames(se), 6, 7)
table(tss)
center <- substr(colnames(se), 27, 28)
table(center)
plate <- substr(colnames(se), 22, 25)
table(plate)
portionanalyte <- substr(colnames(se), 18, 20)
table(portionanalyte)
samplevial <- substr(colnames(se), 14, 16)
table(samplevial)
```

From this information we can make the following observations:

  * All samples were sequenced at the same center

  * All samples belong to one of two combinations of tissue type and vial, matching the
    expected tumor and normal distribution, vials 01A and 01B correspond to tumor samples and 11A and 11B correspond to normal samples.

  * Samples were collected across different tissue source sites (TSS).
  
We are going to use the TSS as surrogate of batch effect indicator. Considering our outcome
of interest as molecular changes between sample types, tumor vs. normal, we will examine now
the cross-classification of this outcome with TSS.

```{r}
table(data.frame(TYPE=se$type, TSS=tss))
```

We see that the samples of each type are perfectly balanced over the TSS, so we anticipate that we will not have any batch effect. We will examine how the samples cluster anyway to ensure that we are right.

We examine now how samples group together by hierarchical clustering and multidimensional
scaling, annotating the outcome of interest and the the surrogate of batch indicator. We
calculate again log CPM values with a higher prior count to moderate extreme fold-changes
produced by low counts. The resulting dendrogram is shown in Figure S6.

```{r sampleClustering, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S6: Hierarchical clustering of the samples."}
logCPM <- cpm(dge, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(tss))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(se)
outcome <- paste(substr(colnames(se), 9, 12), as.character(se$type), sep="-")
names(outcome) <- colnames(se)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)

plot(sampleDendrogram, main="Hierarchical clustering of samples",cex.lab=1.5, cex.axis=1.5, cex.main=1.5)
legend("topright", paste(levels(factor(tss))), fill=sort(unique(batch)))
```

We can observe that samples cluster primarily by sample type, tumor or normal. TSS seems to have a stronger effect among the normal samples, while it distributes better among the tumor samples. We may consider discarding samples that do not seem to cluster as well across batches.

In Figure S7 we show the corresponding MDS plot. Here we see more clearly that the first source of variation separates tumor from normal samples. We can also observe that a few normal samples are separated from the rest, in a more clear separation than that shown in the hierarchical clustering. We may consider discarding those few samples and doing the MDS plot again to have a closer look to the differences among the rest of the samples and their relationship with TSS. In both figures S6 and S7 there a few samples that seem to deviate, but it is not a major deviation, so we decide to not exclude and continue to explore our data. Figures S6 and S7 are created by using TSS as surrogate variable, but we have done the same using the plate, analyte and vial as surrogate variables one at a time and have found similar results, since those plots do not contribute with relevant insight, they are omitted from this report.

```{r mds, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S7: Multidimensional scaling plot of the samples."}

plotMDS(dge, labels=outcome, col=batch,cex.lab=1.5, cex.axis=1.5, cex.main=1.5)
legend("bottomleft", paste(levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05, horiz=TRUE, cex=1.3)
```

## Differential expression

We perform a simple examination of expression changes and their associated p-values
using the R/Bioconductor package [sva](http://bioconductor.org/packages/sva).

```{r}
library(sva)
mod <- model.matrix(~ se$type, colData(se))
mod0 <- model.matrix(~ 1, colData(se))
pv <- f.pvalue(assays(se)$logCPM, mod, mod0)
sum(p.adjust(pv, method="fdr") < 0.01)
```

There are `r sum(p.adjust(pv, method="fdr") < 0.01)` genes changing significantly
their expression at FDR < 1%. In Figure S8 below we show the distribution of the
resulting p-values.

```{r pdist, echo=FALSE, out.width="400px", fig.cap="Figure S8: Distribution of raw p-values for an F-test on every gene between tumor and normal samples."}
hist(pv, main="", las=1, xlab="p-values", cex.lab=1.3, cex.axis=1.1, cex.main=1.3)
```
We see how a vast majority of our genes have a low p-value, to check the distribution of the rest of genes we will plot the histogram of the not significantly expressed genes, this histogram should correspond more or less to a uniform distribution.

```{r pdistzoom, echo=FALSE, out.width="400px", fig.cap="Figure S9: Distribution of p-values which are not significant."}
par(mar = c(4, 8, 3, 1))
hist(pv[pv>0.05], main="", las=1, xlab="p-values", cex.lab=1.3, cex.axis=1.1, cex.main=1.3)
```

The p-values are not perfectly uniform, although it does not seem too bad. Now, let's estimate surrogate variables using the `sva()` function.

```{r}
sv <- sva(assays(se)$logCPM, mod, mod0)
sv$n
```

The SVA algorithm has found `r sv$n` surrogate variables. Let's use them to
assess again the extent of differential expression this time adjusting for these
surrogate variables.

```{r}
modsv <- cbind(mod, sv$sv)
mod0sv <- cbind(mod0, sv$sv)
pvsv <- f.pvalue(assays(se)$logCPM, modsv, mod0sv)
sum(p.adjust(pvsv, method="fdr") < 0.01)
```

We have increased the number of changing genes to `r sum(p.adjust(pvsv, method="fdr") < 0.01)`.
Figure S10 shows the resulting distribution of p-values.

```{r psvdist, echo=FALSE, out.width="400px", fig.cap="Figure S10: Distribution of raw p-values for an F-test on every gene between tumor and normal samples, adjusting for surrogate variables estimated with SVA."}
hist(pvsv, main="", las=1, xlab="p-values after adjusting", cex.lab=1.3, cex.axis=1.1, cex.main=1.3)
```

Once again, we see how a vast majority of our genes have a low p-value, to check the distribution of the rest of genes we will plot the histogram of the not significantly expressed genes.

```{r psvdistzoom, echo=FALSE, out.width="400px", fig.cap="Figure S11: Distribution of p-values which are not significant after adjusting for surrogate variables estimated with SVA."}
hist(pvsv[pvsv>0.05], main="", xlab="p-values after adjusting", las=1, cex.lab=1.3, cex.axis=1.1, cex.main=1.3)
```

The p-values are not perfectly uniform, but the distribution seem a bit more uniform after adjusting.

## Conclusion

The different QA diagnostics reveal some potentially problematic features in some of the samples. Some of this feature may be the difference in library size or variability that is not due to the sample type, as indicated by the MDS plot. However, these issues do not seem overly important, so we have decided to proceed with the analysis without eliminating any more samples after sub-setting, although we may consider discarding them in the future. 

The main source of variation in this data seems to be driven by the tumor and normal condition of the samples. We have found no batch effect using the information from the TGCA barcode. 

The extent of expression changes can be augmented when adjusting for surrogate variables estimated with SVA. Furthermore, the p-values that are not significant are distributed reasonably uniformly, and the distribution seems to improve after the SVA. Additionally, it would be interesting to observe how that extent changes when discarding potentially problematic samples.

## Session information

```{r}
sessionInfo()
```
